File: app/src/main/java/com/dewonderstruck/apps/ashx0/viewmodel/aboutus/AboutUsViewModel.java
package com.dewonderstruck.apps.ashx0.viewmodel.aboutus;

import com.dewonderstruck.apps.Config;
import com.dewonderstruck.apps.ashx0.repository.aboutus.AboutUsRepository;
import com.dewonderstruck.apps.ashx0.utils.AbsentLiveData;
import com.dewonderstruck.apps.ashx0.viewmodel.common.PSViewModel;
import com.dewonderstruck.apps.ashx0.viewobject.AboutUs;
import com.dewonderstruck.apps.ashx0.viewobject.common.Resource;

import javax.inject.Inject;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;

/**
 * Created by Vamsi Madduluri on 12/30/17.
 * Contact Email : vamsii.wrkhost@gmail.com
 */

public class AboutUsViewModel extends PSViewModel {


    //region Variables

    // Get AboutUs
    private final LiveData<Resource<AboutUs>> aboutUsLiveData;
    private MutableLiveData<String> aboutUsObj = new MutableLiveData<>();

    public String aboutId;
    //endregion


    //region Constructors

    @Inject
    AboutUsViewModel(AboutUsRepository repository) {

        aboutUsLiveData = Transformations.switchMap(aboutUsObj, newsId -> {
            if (newsId.isEmpty()) {
                return AbsentLiveData.create();
            }
            return repository.getAboutUs(Config.API_KEY);
        });

    }

    //endregion


    //region Public Methods

    public void setAboutUsObj(String aboutUsObj) {
        this.aboutUsObj.setValue(aboutUsObj);
    }

    public LiveData<Resource<AboutUs>> getAboutUsData() {
        return aboutUsLiveData;
    }

    //endregion

}

File: app/src/main/java/com/dewonderstruck/apps/ashx0/viewmodel/aboutus/AboutUsViewModel.kt
package com.dewonderstruck.apps.ashx0.viewmodel.aboutus

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import com.dewonderstruck.apps.Config
import com.dewonderstruck.apps.ashx0.repository.aboutus.AboutUsRepository
import com.dewonderstruck.apps.ashx0.utils.AbsentLiveData
import com.dewonderstruck.apps.ashx0.viewmodel.common.PSViewModel
import com.dewonderstruck.apps.ashx0.viewobject.AboutUs
import com.dewonderstruck.apps.ashx0.viewobject.common.Resource
import javax.inject.Inject

/**
 * Created by Vamsi Madduluri on 12/30/17.
 * Contact Email : vamsii.wrkhost@gmail.com
 */
class AboutUsViewModel @Inject internal constructor(repository: AboutUsRepository) : PSViewModel() {
    //endregion
    //region Variables
    // Get AboutUs
    val aboutUsData: LiveData<Resource<AboutUs>>
    private val aboutUsObj = MutableLiveData<String>()
    var aboutId: String? = null

    //endregion
    //region Public Methods
    fun setAboutUsObj(aboutUsObj: String) {
        this.aboutUsObj.value = aboutUsObj
    }

    //endregion
    //region Constructors
    init {
        aboutUsData = Transformations.switchMap(aboutUsObj) { newsId: String ->
            if (newsId.isEmpty()) {
                return@switchMap AbsentLiveData.create<Resource<AboutUs>>()
            }
            repository.getAboutUs(Config.API_KEY)
        }
    }
}